bugbug/models/bug_type_classification.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import xgboost
+import numpy as np
+
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.multiclass import OneVsRestClassifier
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup, labels, utils
+from bugbug.model import BugModel
+
+
+# param mode if mode==False, it uses general label names
+#            elif mode==True, it uses the label names with the subcategories
+class BugTypeClassificationModel(BugModel):
+    def __init__(self, lemmatization=False, mode=False, single_class=None):
+        BugModel.__init__(self, lemmatization)
+
+        # These are all the  categories in our taxonomy
+        # both the general and
+        # the sub-categories
+        self.label_names = [
+            'API issue',
+            'Network Usage issue',
+            'Database-related issue',
+            'GUI-related issue',
+            'Performance issue',
+            'Permission-Deprecation issue',
+            'Security issue',
+            'Program Anomaly issue',
+            'Development issue'
+        ]
+
+        label_names_sub_classes = [
+            'Add-on or plug-in incompatibility',
+            'Web incompatibility',
+            'Incompatibility',
+            'Crash',
+            'Hang',
+            'Wrong Functionality',
+            'Incorrect Rendering',
+            'Test code',
+            'Compile'
+        ]
+
+
+
+        # Set the mode  ( General VS Subcategory )
+        self.mode = mode
+
+        # Set the class mode ( Single class VS All the category )
+        self.single_class = single_class
+
+        # Add the sub-classes to the classification list
+        if self.mode == True:
+            self.label_names = self.label_names + label_names_sub_classes
+
+        # Check if the element is based over the single class
+        if self.single_class is not None:
+            assert self.single_class is not None, f"unexpected class\n\tUse the args --single_class with one of the following {self.label_names}"
```

Comment:
Nit: this assert always passes, since you are asserting the same condition of the if